fe1-web/src/features/notification/reducer/NotificationReducer.ts

```diff
@@ -0,0 +1,201 @@
+/**
+ * This error is disabled since reducers use the createSlice function, which requires the user to
+ * param-reassign. Please do not disable other errors.
+ */
+/* eslint-disable no-param-reassign */
+import { createSelector, createSlice, Draft, PayloadAction } from '@reduxjs/toolkit';
+
+export interface NotificationState {
+  /* the id of the notification, is automatically assigned */
+  id: number;
+  /* the id of the lao this notification is associated with */
+  laoId: string;
+  /* whether the notification has been read. is automatically assigned */
+  hasBeenRead: boolean;
+  /* the time associated with the notification */
+  timestamp: number;
+  /* the title that is shown in the notification */
+  title: string;
+  /* this field can be used to differentiate various types of notifications */
+  type: string;
+}
+
+export const NOTIFICATION_REDUCER_PATH = 'notifications';
+
+export interface NotificationReducerState {
+  byLaoId: {
+    [laoId: string]: { byId: Record<number, NotificationState>; allIds: number[]; nextId: number };
+  };
+}
+
+const initialState: NotificationReducerState = {
+  byLaoId: {},
+};
+
+const notificationSlice = createSlice({
+  name: NOTIFICATION_REDUCER_PATH,
+  initialState,
+  reducers: {
+    // Action called to add a new notification
+    addNotification: (
+      state: Draft<NotificationReducerState>,
+      action: PayloadAction<Omit<NotificationState, 'id' | 'hasBeenRead'>>,
+    ) => {
+      // check if there are already notifications for this lao
+      let laoState = state.byLaoId[action.payload.laoId];
+      if (!laoState) {
+        // if not create a new, empty state for that lao
+        laoState = {
+          byId: {},
+          allIds: [],
+          nextId: 0,
+        };
+      }
+
+      const notification: NotificationState = {
+        ...action.payload,
+        id: laoState.nextId,
+        hasBeenRead: false,
+      };
+
+      laoState.nextId += 1;
```

Comment:
You do highlight a very important problem, which I believe doesn't affect us because... JavaScript !

JavaScript is not strictly concurrent, it's runtime model is based on an event loop.
Cf. https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop
Furthermore, reducers are basically called synchronously. 
Effectively, this means that there's only ever going to be one `addNotification` being executed.

Reply:
Ah and the event loop ["runs to completion"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop#run-to-completion). Thank you for the pointer it is good to have a reference for this! 😄